obspy/clients/filesystem/tsindex.py

```diff
@@ -0,0 +1,1820 @@
+            :class:`~TSIndexDatabaseHandler`
+        :param database: Path to sqlite tsindex database or a
+            TSIndexDatabaseHandler object
+        :type datapath_replace: 2-value tuple(str, str)
+        :param datapath_replace: A 2-value ``tuple(str, str)``, where any
+            occurrence of the first value will be replaced with the second
+            value in filename paths from the index.
+        :type loglevel: str
+        :param loglevel: logging verbosity
+        """
+        numeric_level = getattr(logging, loglevel.upper(), None)
+        if not isinstance(numeric_level, int):
+            raise ValueError('Invalid log level: %s' % loglevel)
+        logging.basicConfig(level=numeric_level)
+        logger.setLevel(numeric_level)
+
+        # setup handler for database
+        if isinstance(database, (str, native_str)):
+            self.request_handler = TSIndexDatabaseHandler(
+                os.path.normpath(database),
+                loglevel=loglevel)
+        elif isinstance(database, TSIndexDatabaseHandler):
+            self.request_handler = database
+        else:
+            raise ValueError("database must be a string or "
+                             "TSIndexDatabaseHandler object.")
+
+        # Create and configure the data extraction
+        self.data_extractor = _MiniseedDataExtractor(
+            dp_replace=datapath_replace,
+            loglevel=loglevel)
+
+    def get_waveforms(self, network, station, location,
+                      channel, starttime, endtime, merge=-1):
+        """
+        Query tsindex database and read miniSEED data from local indexed
+        directory tree.
+
+        :type network: str
+        :param network: Network code of requested data (e.g. "IU").
+            Wildcards '*' and '?' are supported.
+        :type station: str
+        :param station: Station code of requested data (e.g. "ANMO").
+            Wildcards '*' and '?' are supported.
+        :type location: str
+        :param location: Location code of requested data (e.g. "").
+            Wildcards '*' and '?' are supported.
+        :type channel: str
+        :param channel: Channel code of requested data (e.g. "HHZ").
+            Wildcards '*' and '?' are supported.
+        :type starttime: :class:`~obspy.core.utcdatetime.UTCDateTime`
+        :param starttime: Start of requested time window.
+        :type endtime: :class:`~obspy.core.utcdatetime.UTCDateTime`
+        :param endtime: End of requested time window.
+        :type merge: int or None
+        :param merge: Specifies, which merge operation should be performed
+            on the stream before returning the data. Default (``-1``) means
+            only a conservative cleanup merge is performed to merge seamless
+            traces (e.g. when reading across day boundaries). See
+            :meth:`Stream.merge() <obspy.core.stream.Stream.merge>` for
+            details. If set to ``None`` (or ``False``) no merge operation at
```

Comment:
does this mean that all options/args/kwargs passed to ``Stream.merge()`` are valid here ? like ``Stream.merge(method=1, fill_value="interpolate")`` etc ?

Reply:
This was a direct copy from `~obspy.clients.filesystem.sds`: https://github.com/obspy/obspy/blob/master/obspy/clients/filesystem/sds.py#L146

Reply:
So @megies knows :-)